src/index.js

```diff
@@ -1,10 +1,67 @@
 'use strict';
 
 const path = require('path');
+const fs = require('fs');
+
 const requireIndex = require('requireindex');
+const chromeLauncher = require('chrome-launcher');
+const lighthouse = require('lighthouse');
+const isURL = require('is-url');
+const Adviser = require('adviser');
+
+class LighthousePlugin extends Adviser.Plugin {
+  constructor(settings) {
+    super(settings);
+
+    if (!Object.prototype.hasOwnProperty.call(settings, 'url') || !isURL(settings.url)) {
+      throw new Error(`No valid url provided.`);
+    }
+
+    this.url = settings.url;
+    this.options = settings.options || {};
+    this.configPath = settings.configPath;
+    this.rules = requireIndex(path.join(__dirname, '/rules'));
+  }
+
+  async preRun(context) {
+    let config = null;
+
+    if (this.configPath) {
+      const fullConfigPath = path.join(context.filesystem.dirname, this.configPath);
+
+      if (!fs.existsSync(fullConfigPath)) {
```

Comment:
Won't the try catch below catch this error as well?

Reply:
It doesn't, it was exploding